config/stack.go

```diff
@@ -183,45 +190,55 @@ func generateStackFile(ctx context.Context, opts *options.TerragruntOptions, sta
 // generateUnits iterates through a slice of Unit objects, processing each one by copying
 // source files to their destination paths and writing unit-specific values.
 // It logs the processing progress and returns any errors encountered during the operation.
-func generateUnits(ctx context.Context, opts *options.TerragruntOptions, sourceDir, targetDir string, units []*Unit) error {
-	for _, unit := range units {
-		item := itemToProcess{
-			sourceDir: sourceDir,
-			targetDir: targetDir,
-			name:      unit.Name,
-			path:      unit.Path,
-			source:    unit.Source,
-			values:    unit.Values,
-		}
+func generateUnits(ctx context.Context, opts *options.TerragruntOptions, pool *util.WorkerPool, sourceDir, targetDir string, units []*Unit) error {
+	for _, u := range units {
+		pool.Submit(func() error {
+			unit := u
```

Comment:
Is this necessary for some reason?

Reply:
I think yes, I want to have a copy to avoid capturing the loop variable reference

Reply:
I believe you don't have to do that anymore in modern go, but this might be a special case